### PR TITLE
Use Gradle 7.6 to build Gradle plugin

### PR DESCRIPTION
**Pull request by erichaagdev**
This PR updates the `openapi-generator-gradle-plugin` project to build with Gradle 7.6, the latest. 

All plugin versions have been updated to their latest versions. The plugins `io.codearte.nexus-staging` and `de.marcphilipp.nexus-publish` have been replaced by [`io.github.gradle-nexus.publish-plugin`](https://github.com/gradle-nexus/publish-plugin/) as recommended by the original plugin authors. Aside from the configuration, the only notable difference is that "Nexus" has been replaced with "Sonatype" in the names of the relevant publish tasks.

I'm not sure why previously there existed both a `mavenJava` and `pluginMaven` publication. For a Gradle plugin, the only publication needed is `pluginMaven` created by the `java-gradle-plugin` plugin. I've removed the `mavenJava` publication. Only `pluginMaven` is published to Sonatype. 

I am unable to test publishing to Sonatype. However, I compared the before and after output of `publishToMavenLocal` and saw no notable differences.

I did not update the sample project as it does not depend on the version of Gradle used to build the plugin itself. I also did not want to make this PR any larger.

Gradle Build Scan: https://gradle.com/s/d4jzctozh7gem
Gradle 7.6 Release Notes: https://docs.gradle.org/7.6/release-notes.html

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -160,24 +160,24 @@ after_success:
         echo "Finished mvn clean deploy for $TRAVIS_BRANCH";
         pushd .;
         cd modules/openapi-generator-gradle-plugin;
-        ./gradlew -Psigning.keyId="$SIGNING_KEY" -Psigning.password="$SIGNING_PASSPHRASE" -Psigning.secretKeyRingFile="${TRAVIS_BUILD_DIR}/sec.gpg" -PossrhUsername="${SONATYPE_USERNAME}" -PossrhPassword="${SONATYPE_PASSWORD}" publishMavenJavaPublicationToNexusRepository closeAndReleaseRepository --no-daemon;
-        echo "Finished ./gradlew publishPluginMavenPublicationToNexusRepository closeAndReleaseRepository";
+        ./gradlew -Psigning.keyId="$SIGNING_KEY" -Psigning.password="$SIGNING_PASSPHRASE" -Psigning.secretKeyRingFile="${TRAVIS_BUILD_DIR}/sec.gpg" -PossrhUsername="${SONATYPE_USERNAME}" -PossrhPassword="${SONATYPE_PASSWORD}" publishPluginMavenPublicationToSonatypeRepository closeAndReleaseRepository;
```
Comment (erichaagdev) [inline .travis.yml:163]: I could not figure out why previously the `publishMavenJavaPublicationToNexusRepository` task was called in the first `if` condition, but the `publishPluginMavenPublicationToNexusRepository` task was called in the second. I came to the conclusion this was probably not intentional.

In any case, there is no longer a `mavenJava` publication so `publishPluginMavenPublicationToSonatypeRepository` is called in both places.

**Comment by erichaagdev** [comment]
@wing328 could you please review this PR? Thank you!

**Comment by erichaagdev** [comment]
@sorin-florea I'd be honored to have you review this PR as well. :smile:

**Comment by erichaagdev** [comment]
@saschpe a review from you would be appreciated too.

**Comment by erichaagdev** [comment]
@cbornet @jmini @etherealjoy @spacether could I get a review? Thank you!

**Comment by wing328** [comment]
@erichaagdev thanks for the PR. Can you please PM me via Slack for a few quick questions when you've time?

**Review by sorin-florea — APPROVED** [review]

File: shippable.yml [hunk 2]
```diff
@@ -14,11 +14,11 @@ build:
...
+    # install Gradle 7.6
+    - wget https://services.gradle.org/distributions/gradle-7.6-bin.zip
     - sudo mkdir /opt/gradle
-    - unzip -d /opt/gradle gradle-5.6.4-bin.zip
-    - export PATH=/opt/gradle/gradle-5.6.4/bin:$PATH 
+    - unzip -d /opt/gradle gradle-7.6-bin.zip
+    - export PATH=/opt/gradle/gradle-7.6/bin:$PATH
     - gradle -v
     - java -version
     # ensure all modifications created by 'mature' generators are in the git repo
```
Comment (wing328) [inline shippable.yml:24]: FYI. thanks for updating it but we no longer use Shippable CI. 

I'll file another PR to clean up this file later.

**Comment by wing328** [comment]
CircleCI node 1 failure not related to this change.

File: modules/openapi-generator-gradle-plugin/build.gradle [hunk 3]
```diff
@@ -1,84 +1,56 @@
...
-targetCompatibility = 1.8
+java {
+    withSourcesJar()
+    withJavadocJar()
+    sourceCompatibility = 1.8
+    targetCompatibility = 1.8
+}
 
 repositories {
-    mavenLocal()
```
Comment (wing328) [inline modules/openapi-generator-gradle-plugin/build.gradle:48]: @erichaagdev I think we will need to add this back. Please take a look at the build failure in this PR: https://github.com/OpenAPITools/openapi-generator/pull/14172. 

After adding back this, I can no longer repeat the issue locally via "mvn clean install"

Comment (erichaagdev) [reply modules/openapi-generator-gradle-plugin/build.gradle:48]: Ah, I do think it should be there. Good catch. Do you want me to add it back or will you add it in your PR?

Comment (erichaagdev) [reply modules/openapi-generator-gradle-plugin/build.gradle:48]: However the PR failure I am looking at seems to be related to connectivity, not about resolving dependencies from Maven local.

https://github.com/OpenAPITools/openapi-generator/actions/runs/3620526865/jobs/6102851542#step:5:114

Comment (wing328) [reply modules/openapi-generator-gradle-plugin/build.gradle:48]:
```
> Task :publishOpenApiGeneratorPluginMarkerMavenPublicationToMavenLocal
> Task :generatePomFileForPluginMavenPublication

> Task :compileKotlin FAILED
e: /home/runner/work/openapi-generator/openapi-generator/modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/tasks/GenerateTask.kt: (770, 34): Unresolved reference: addOpenAPINormalizer
```

We added a new method `addOpenAPINormalizer` and it's only available in the local maven cache store. 

I already filed https://github.com/OpenAPITools/openapi-generator/pull/14265

Comment (wing328) [reply modules/openapi-generator-gradle-plugin/build.gradle:48]: UPDATE: merged #14265

File: .travis.yml [hunk 4]
```diff
@@ -160,24 +160,24 @@ after_success:
...
+        ./gradlew -PossrhUsername="${SONATYPE_USERNAME}" -PossrhPassword="${SONATYPE_PASSWORD}" publishPluginMavenPublicationToSonatypeRepository closeAndReleaseRepository;
+        echo "Finished ./gradlew publishPluginMavenPublicationToSonatypeRepository closeAndReleaseRepository";
         popd;
       fi;
       if [ -n $TRAVIS_TAG ] && [[ "$TRAVIS_TAG" =~ ^[v][0-9]+\.[0-9]+\.[0-9]+$ ]]; then
         echo "Publishing the gradle plugin to Gradle Portal on tag $TRAVIS_TAG (only)";
         pushd .;
         cd modules/openapi-generator-gradle-plugin;
-        ./gradlew -Psigning.keyId="$SIGNING_KEY" -Psigning.password="$SIGNING_PASSPHRASE" -Psigning.secretKeyRingFile="${TRAVIS_BUILD_DIR}/sec.gpg" publishPlugins -Dgradle.publish.key=$GRADLE_PUBLISH_KEY -Dgradle.publish.secret=$GRADLE_PUBLISH_SECRET --no-daemon;
+        ./gradlew -Psigning.keyId="$SIGNING_KEY" -Psigning.password="$SIGNING_PASSPHRASE" -Psigning.secretKeyRingFile="${TRAVIS_BUILD_DIR}/sec.gpg" publishPlugins -Dgradle.publish.key=$GRADLE_PUBLISH_KEY -Dgradle.publish.secret=$GRADLE_PUBLISH_SECRET;
```
Comment (wing328) [inline .travis.yml:180]: @erichaagdev we're working on v6.3.0 release but Travis CI shows the following errors:
```
For more details see https://docs.gradle.org/7.6/release-notes.html


> Configure project :
Signing plugin detected. Will automatically sign the published artifacts.

> Task :pluginDescriptors UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :sourcesJar UP-TO-DATE
> Task :generatePomFileForPluginMavenPublication
> Task :generatePomFileForOpenApiGeneratorPluginMarkerMavenPublication
> Task :signOpenApiGeneratorPluginMarkerMavenPublication FAILED

> Task :compileKotlin
w: /home/travis/build/OpenAPITools/openapi-generator/modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/tasks/GenerateTask.kt: (805, 48): 'getter for config: CodegenConfig!' is deprecated. Deprecated in Java
w: /home/travis/build/OpenAPITools/openapi-generator/modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/tasks/GenerateTask.kt: (811, 40): 'getter for config: CodegenConfig!' is deprecated. Deprecated in Java

FAILURE: Build failed with an exception.

* What went wrong:
Could not evaluate onlyIf predicate for task ':signOpenApiGeneratorPluginMarkerMavenPublication'.
> Could not evaluate spec for 'Signing is required, or signatory is set'.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.6/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 9s
7 actionable tasks: 4 executed, 3 up-to-date
```

Do you know what may have caused this?

Comment (wing328) [reply .travis.yml:180]: Travis CI job full log: https://app.travis-ci.com/github/OpenAPITools/openapi-generator/jobs/594874885

**Pull request merged**